api/pkg/artifact/artifact.go

```diff
@@ -23,44 +40,38 @@ type URL struct {
 	Object string
 }
 
-type Service interface {
-	ParseURL(gsURL string) (*URL, error)
-
-	ReadArtifact(ctx context.Context, url string) ([]byte, error)
-	WriteArtifact(ctx context.Context, url string, content []byte) error
-	DeleteArtifact(ctx context.Context, url string) error
-}
+type URLScheme string
 
-type GcsArtifactClient struct {
-	API *storage.Client
+type SchemeInterface interface {
+	GetURLScheme() string
+	ParseURL(gsURL string) (*URL, error)
 }
```

Comment:
Creating a new interface for the artifact scheme allows us to implement the interface methods `GetURLScheme` and `ParseURL` only once and inherit it directly within all the implementations of `Service` (instead of implementing them each time we have a new implementation of `Service`).